### PR TITLE
mod: Seperate ranged damage values

### PR DESCRIPTION
**Pull request by Norr2**

### CODE REVIEW

File: src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs [hunk 1]
```diff
@@ -127,61 +127,121 @@ public CrpgAgentApplyDamageModel(CrpgConstants constants)
...
-            case BoneBodyPartType.Legs:
-                result = 0.8f;
-                break;
+        if (isMissile)
+        {
+            switch (bodyPart)
+            {
+                case BoneBodyPartType.None:
+                    result = 1f;
+                    break;
```
Comment (namidaka) [inline src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs:140]: The code remains insanely nested.
What i recommend : 
CalculateMeleeDamageMultiplierForHumanBodyPart
CalculateMeleeDamageMultiplierForNonHumanBodyPart
CalculateRangedDamageMultiplierForHumanBodyPart
CalculateRangedDamageMultiplierForNonHumanBodyPart

Then 
If(IsMissile)
{
   return IsHuman ? CalculateRangedDamageMultiplierForHumanBodyPart(ArgumentsGoHere) : CalculateRangedDamageMultiplierForNonHumanBodyPart(ArgumentsGoHere)
}
else
{
 return IsHuman ? CalculateMeleeDamageMultiplierForHumanBodyPart(ArgumentsGoHere) : CalculateMeleeDamageMultiplierForNonHumanBodyPart(ArgumentsGoHere)
}

this should kill any form of nesting

**Review by namidaka — CHANGES_REQUESTED** [review]

File: src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs (hunk 1 above)
Comment (Norr2) [reply src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs]: Attempted 2nd draft, checking it is what you meant before continuing
I feel like it is wrong still, would add 4 more methods, increasing code, but would make
"public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart, DamageTypes type, bool isHuman, bool isMissile)" method short

Comment (Norr2) [reply src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs]: Unless you also mean to rewrite the TW base code with lots of switches etc?

Comment (namidaka) [reply src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs]: > Attempted 2nd draft, checking it is what you meant before continuing I feel like it is wrong still, would add 4 more methods, increasing code, but would make "public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart, DamageTypes type, bool isHuman, bool isMissile)" method short

Don't worry , we don't have to pay any extra for the extra lines of code.  
You spend 20% of the time writing code , and 80% trying to figure what the hell it means when you come back to it 6 month later. Good code is code that you can understand as you read it. If you have to follow nested condition and remember everytime what branch you are in it makes it harder for you to read. Here's a good short video on the subject
https://www.youtube.com/watch?v=CFRhGnuXG-4

Comment (Norr2) [reply src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs]: ok, updated and finished, + cleaned it up a little

File: src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs [hunk 2]
```diff
@@ -177,7 +216,93 @@ public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart,
...
+                {
+                    case DamageTypes.Invalid:
+                        result = 2f;
+                        break;
+                    case DamageTypes.Cut:
+                        result = 1.2f;
+                        break;
+                    case DamageTypes.Pierce:
+                        result = 1.6f;
+                        break;
```
Comment (namidaka) [inline src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs:249]: This was 1.3 before the patch as the calculation was done in another file before

**Review by namidaka — CHANGES_REQUESTED** [review]

File: src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs (hunk 2 above)
Comment (Norr2) [reply src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs:249]: I have the second PR ready to make changes to values as requested, this PR was just to improve the format, reduce nesting

**Review by namidaka — APPROVED** [review]

**Pull request merged**